diff --git a/hardware/qcom/display/libhwcomposer/hwcomposer.cpp b/hardware/qcom/display/libhwcomposer/hwcomposer.cpp
index ad0acc6..03bf90b 100644
--- a/hardware/qcom/display/libhwcomposer/hwcomposer.cpp
+++ b/hardware/qcom/display/libhwcomposer/hwcomposer.cpp
@@ -493,7 +493,7 @@ bool isMDPSupported( const hwc_layer_list_t* list ) {
 
         if(((src_w > dst_w) || (src_h > dst_h)) && needsBlending){
 
-            if(not FrameBufferInfo::getInstance()->canSupportTrueMirroring()){
+            if(not overlay::FrameBufferInfo::getInstance()->canSupportTrueMirroring()){
                 /* If the target MDP version is less than 4.2, we
                  * cannot handle alpha downscaling. In such cases bypass is
                  * not possible and we return false.*/
