diff --git a/system/netd/Android.mk b/system/netd/Android.mk
index c09b634..d67a307 100644
--- a/system/netd/Android.mk
+++ b/system/netd/Android.mk
@@ -36,14 +36,21 @@ LOCAL_CFLAGS :=
 LOCAL_SHARED_LIBRARIES := libstlport libsysutils libcutils libnetutils \
                           libcrypto libhardware_legacy
 
+#backward compatibility
+ifeq ($(WIFI_DRIVER_MODULE_NAME),ar6000)
+LOCAL_CFLAGS += -DWIFI_MODULE_PATH=\"$(WIFI_DRIVER_MODULE_PATH)\"
+LOCAL_C_INCLUDES += external/wpa_supplicant external/hostapd
+LOCAL_SRC_FILES += SoftapControllerATH.cpp
+LOCAL_SHARED_LIBRARIES := $(LOCAL_SHARED_LIBRARIES) libwpa_client
+else
 ifdef BOARD_SOFTAP_DEVICE_TI
     LOCAL_SRC_FILES += SoftapControllerTI.cpp
     LOCAL_C_INCLUDES += external/libnl-headers
     LOCAL_STATIC_LIBRARIES += libnl_2
 else
     LOCAL_SRC_FILES += SoftapController.cpp
-endif
-
+endif #BOARD_SOFTAP_DEVICE_TI
+endif #WIFI_DRIVER_MODULE_NAME==ar6000
 
 ifneq ($(BOARD_HOSTAPD_DRIVER),)
   LOCAL_CFLAGS += -DHAVE_HOSTAPD
